src/STW.ProcessingApi.Function/Triggers/ProcessingApiTrigger.cs

```
@@ -19,12 +19,31 @@ public ProcessingApiTrigger(ILogger<ProcessingApiTrigger> logger, IValidationSer
     }
 
     [Function(nameof(ProcessingApiTrigger))]
-    public async Task Run([ServiceBusTrigger("%ServiceBusQueueName%", Connection = "ServiceBusConnectionString")] ServiceBusReceivedMessage message)
+    public async Task Run(
+        [ServiceBusTrigger("%ServiceBusQueueName%", Connection = "ServiceBusConnectionString")]
+        ServiceBusReceivedMessage message)
     {
         _logger.LogInformation($"{nameof(ProcessingApiTrigger)} function was invoked.");
 
-        var spsCertificate = JsonSerializer.Deserialize<SpsCertificate>(message.Body);
+        var messageBody = message.Body.ToString();
 
-        await _validationService.InvokeRulesAsync(spsCertificate!);
+        try
+        {
+            var spsCertificate = JsonConvert.DeserializeObject<SpsCertificate>(messageBody);
+
+            var errors = await _validationService.InvokeRulesAsync(spsCertificate!);
+
+            if (errors.Count == 0)
+            {
+                _logger.LogInformation("Validation passed");
+            }
```

The validation passed/failed logic would be better placed in the validation service rather than here.

I don't think we will need to catch the JSON exception and should let it bubble up as it will be visible in the Azure portal.

---

Feels like it makes more sense to have the handling of the errors here for when they get sent to another service so we can expose them to STW? I'll change back for now